### DISCUSSION TITLE
Production Docker image 404 assets

### DISCUSSION THREAD

**Discussion by guizaols** [ElMassimo/vite_ruby#173] [Q&A] [answered]
Hello everybody.

I'm building a Docker image (Ruby Alpine 3.0.3) to my production environment and the assets are been generated nicely. But when I access the application I get only 404 error to the assets even with them in the directories.

Is there any idea or tip about what's going on or something to be changed over here?

PS: I deployed another app this week in Heroku and the things were perfectly.

![Screen Shot 2022-01-10 at 18 12 53](https://user-images.githubusercontent.com/1687999/148840270-08ada0f7-d68c-44ac-9b8b-c3be3cf29b5b.png)

**Comment by ElMassimo** [comment, marked as answer]
Hi!

Heroku sets [an environment variable](https://github.com/ElMassimo/vite_ruby/blob/main/examples/rails/config/environments/production.rb#L24) to serve static files in production, by default.

You might want to check this setting is enabled, by setting it to true explicitly in `config/environments/production.rb`, or by setting the environment variable in the Docker image.

**Reply by guizaols** [reply]
Thanks a lot @ElMassimo. I set this `RAILS_SERVE_STATIC_FILES` env and it's working now!
Could you explain why is it needed?

Thanks again!

**Reply by ElMassimo** [reply]
By default, [Rails does not serve static assets](https://edgeguides.rubyonrails.org/configuring.html#config-public-file-server-enabled).

However, when using a third-party CDN service, it can be convenient for the Rails server to respond to the initial requests, which is one of the reasons why it's enabled by default in Heroku. When not using a server like NGINX or a CDN, you would still want to serve assets (not recommended from a performance standpoint, but it works).

**Reply by guizaols** [reply]
Actually I'm running the app through Ingress-Nginx (K8S). So I should take a look in how to serve these assets over there.
Anyway thanks again for your quick answer, your explanation and great job doing this gem!
